File: 55-jump-game/55-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int length = nums.size();
        int reachable = 0;
        for(int i=0;i<length;i++){
            if(reachable < i){
                return false;
            }
            reachable = max(reachable,i+nums[i]);
        }
        return true;
    }
};

File: 55-jump-game/55-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        length = len(nums)
        reachable = 0
        for i in range(length):
            if (reachable < i):
                return False
            reachable = max(reachable,i+nums[i])
        return True